### DISCUSSION TITLE
request params is not support

### DISCUSSION THREAD

**Discussion by yztdemo** [fastapi/fastapi#8716] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Item(BaseModel):  
    name: str
    age: int
    height: float

app = FastAPI()

@app.get("/hh")
async def update_item(item: Item):  # item需要与Item对象定义保持一致
    return {
        "method": 'get',
        "people_name": item.name,
        "people_age": item.age,
        "people_height": item.height
    }
```

### Description

this  request  method    is     **GET**    
 if   i    request  this  function  by    postman  ,
 
![image](https://user-images.githubusercontent.com/62371320/147324107-03fbf03e-ecd3-4803-93a7-2c19fc828279.png)

so   that is  not  support        

### Operating System

Windows

### FastAPI Version

0.63.0

### Python Version

Python 3.10.0

### Additional Context

emo

**Comment by insomnes** [comment]
You have wrong code in your handler, pydantic model is used for body parameters from JSON in POST request.
More on it in [docs](https://fastapi.tiangolo.com/tutorial/body/).
And as you can see error message shows you that you have missing fields in body. 
If you want to use query parameters your code should look like:
```python
@app.get("/hh")
async def update_item(name: str, age: int, height: float):
    return {
        "method": 'get',
        "people_name": item.name,
        "people_age": item.age,
        "people_height": item.height
    }
```
More on query parameters in [docs](https://fastapi.tiangolo.com/tutorial/query-params/).
But judging by the name of the handler function, you should use the POST method and send the JSON inside.

**Comment by panla** [comment]
```python
# to resolve GET method

item: Item = Depends(Item)
```

**Comment by yztdemo** [comment]
> ```python
> # to resolve GET method
> 
> item: Item = Depends(Item)
> ```

thank you    your method  is right        you  is  my  angel baby

**Comment by yztdemo** [comment]
> You have wrong code in your handler, pydantic model is used for body parameters from JSON in POST request. More on it in [docs](https://fastapi.tiangolo.com/tutorial/body/). And as you can see error message shows you that you have missing fields in body. If you want to use query parameters your code should look like:
> 
> ```python
> @app.get("/hh")
> async def update_item(name: str, age: int, height: float):
>     return {
>         "method": 'get',
>         "people_name": item.name,
>         "people_age": item.age,
>         "people_height": item.height
>     }
> ```
> 
> More on query parameters in [docs](https://fastapi.tiangolo.com/tutorial/query-params/). But judging by the name of the handler function, you should use the POST method and send the JSON inside.

i  know  what you say ,   see  the panla  comments,  you will know other way  to  deliver params   of  **GET**

**Comment by insomnes** [comment]
@panla interesting method didn't know about it, thank you!

@yztdemo I must still admit that for clarity if you want to update something it is better to not use GET method, cause it's misleading.
As mentioned in [docs](https://fastapi.tiangolo.com/tutorial/body-updates/) it's better to use PUT or PATCH methods.

**Comment by panla** [comment]
@insomnes   @yztdemo 

- [issue 3525](https://github.com/tiangolo/fastapi/issues/3525)
- [issue 2997](https://github.com/tiangolo/fastapi/issues/2997)
- [classes-as-dependencies](https://fastapi.tiangolo.com/zh/tutorial/dependencies/classes-as-dependencies/)

```python
name: Optional[str] = Query('', title='name')
```

**Comment by YuriiMotov** [comment, marked as answer]
Since FastAPI 0.115.0 you can modify your code by adding `Query` like shown below to make it working:

```
@app.get("/hh")
async def update_item(item: Annotated[Item, Query()]):
    return {
        "method": 'get',
        "people_name": item.name,
        "people_age": item.age,
        "people_height": item.height
    }
```
